modal/sandbox.py

```diff
@@ -0,0 +1,94 @@
+from typing import Optional, Type
+
+from grpclib.exceptions import GRPCError, StreamTerminatedError
+
+from modal_proto import api_pb2
+from modal_utils.async_utils import synchronize_api
+from modal_utils.grpc_utils import RETRYABLE_GRPC_STATUS_CODES, retry_transient_errors, unary_stream
+
+from .client import _Client
+from .object import _Handle
+
+
+class _LogsReader:
```

Comment:
don't we already have very similar code for this elsewhere?

Reply:
yeah mentioned in a comment a few lines below. We can combine later.

Reply:
there's a bunch of small differences that make it not worth the effort of doing now